From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Vincent Strubel <clipos@ssi.gouv.fr>
Date: Mon, 19 Mar 2012 14:20:42 +0100
Subject: [PATCH 22/32] clip: grsec: Reduce GRKERNSEC_BRUTE user ban to 30
 seconds

Réduire l'interdiction d'utilisateur GRKERNSEC_BRUTE de 15 minutes à 30 secondes.

Patch: 1522_grsec_brute_timeout.patch
---
 grsecurity/grsec_sig.c | 4 ++--
 include/linux/grmsg.h  | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/grsecurity/grsec_sig.c b/grsecurity/grsec_sig.c
index f072c9d4291ad..2f7bbcca846dc 100644
--- a/grsecurity/grsec_sig.c
+++ b/grsecurity/grsec_sig.c
@@ -75,8 +75,8 @@ int gr_fake_force_sig(int sig, struct task_struct *t)
 }
 #endif
 
-#define GR_USER_BAN_TIME (15 * 60)
-#define GR_DAEMON_BRUTE_TIME (30 * 60)
+#define GR_USER_BAN_TIME 30
+#define GR_DAEMON_BRUTE_TIME 30
 
 void gr_handle_brute_attach(int dumpable)
 {
diff --git a/include/linux/grmsg.h b/include/linux/grmsg.h
index f1d36054b8e43..990fbbae2cd7b 100644
--- a/include/linux/grmsg.h
+++ b/include/linux/grmsg.h
@@ -115,7 +115,7 @@
 #define GR_BADPROCPID_MSG "denied read of sensitive /proc/pid/%s entry via fd passed across exec by "
 #define GR_SYMLINKOWNER_MSG "denied following symlink %.950s since symlink owner %u does not match target owner %u, by "
 #define GR_BRUTE_DAEMON_MSG "bruteforce prevention initiated for the next 30 minutes or until service restarted, stalling each fork 30 seconds.  Please investigate the crash report for "
-#define GR_BRUTE_SUID_MSG "bruteforce prevention initiated due to crash of %.950s against uid %u, banning suid/sgid execs for %u minutes.  Please investigate the crash report for "
+#define GR_BRUTE_SUID_MSG "bruteforce prevention initiated due to crash of %.950s against uid %u, banning suid/sgid execs for %u seconds.  Please investigate the crash report for "
 #define GR_IPC_DENIED_MSG "denied %s of overly-permissive IPC object with creator uid %u by "
 #define GR_TIOCSTI_MSG "denied unprivileged use of TIOCSTI by "
 #define GR_MSRWRITE_MSG "denied write to CPU MSR by "
-- 
2.17.0

